src/mozillavpn.cpp

```diff
@@ -90,7 +92,18 @@ MozillaVPN::MozillaVPN() : m_private(new Private()) {
   logger.debug() << "Creating MozillaVPN singleton";
 
 #ifdef MVPN_ADJUST
-  AdjustHandler::initialize();
+  AdjustProxy* adjustProxy = new AdjustProxy(qApp);
+  QObject::connect(controller(), &Controller::readyToQuit, adjustProxy,
+                   &AdjustProxy::close);
+  for (int i = 0; i < 5; i++) {
+    quint16 port = QRandomGenerator::global()->bounded(1024, 65536);
+    bool succeeded = adjustProxy->initialize(port);
+    if (succeeded) {
+      break;
+    }
+  }
+  AdjustHandler::initialize(adjustProxy->serverPort());
```

Comment:
if !succeeded, do not activate adjust

Reply:
We check in the `initialize` function if the proxy is listening.